docs/book/src/reference/glossary.md

```diff
@@ -210,6 +240,27 @@ it usually contains Custom Resource Definitions (CRDs), Deployments (to run the
 
 Existing Cluster API implementations consist of generic and infrastructure provider-specific logic. The [infrastructure provider](#infrastructure-provider)-specific logic is currently maintained in infrastructure provider repositories.
 
+# R
+---
+
+### Runtime Extension
+
+An external component which is part of a system built on top of Cluster API that can handle requests for a specific Runtime Hook.
```

Comment:
I think handlers are more clearly on the implementation side and we should limit this doc to high-level vocabulary rather than trying to re-define out API types here.